cocotb/handle.py

```diff
@@ -530,11 +519,67 @@ def _range_iter(self, left, right):
 
     @NonHierarchyObject.value.getter
     def value(self):
-        # need to iterate over the sub-object
-        result = []
-        for x in range(len(self)):
-            result.append(self[x].value)
-        return result
+        """Reference to the value of the simulation object.
```

Comment:
```suggestion
        """A list of each value within this simulation object.
```
`reference` has implications of the write through semantics that don't exist that you warn about below

Reply:
How should we word it for the non-array classes.
Also I'm confused why `ModifiableObject` is a descendant of `NonHierarchyIndexableObject`.

Reply:
> How should we word it for the non-array classes.

Not sure. I assume you're referring to

https://github.com/cocotb/cocotb/blob/8248bd336bdb0a641334f17a18176833428ae6fd/cocotb/handle.py#L391-L394

> Also I'm confused why `ModifiableObject` is a descendant of `NonHierarchyIndexableObject`.

I'm afraid I can't help you there